opennms-config/src/main/java/org/opennms/netmgt/config/DiscoveryConfigFactory.java

```diff
@@ -570,6 +584,39 @@ void initializeExcludeRanges() {
         }
     }
 
+    void initializeExcludeUrls() {
+        getReadLock().lock();
+        List<ExcludeUrl> excludeUrlList = new ArrayList<>();
+        try {
+            excludeUrlList.addAll(getConfiguration().getExcludeUrls());
+            // Add exclude urls from definitions. Assign location from definition.
+            getConfiguration().getDefinitions().forEach(def -> {
+                def.getExcludeUrls().forEach(eu -> {
+                    if (!eu.getLocation().isPresent()) {
+                        eu.setLocation(def.getLocation().orElse(LocationUtils.DEFAULT_LOCATION_NAME));
+                    }
+                });
+                excludeUrlList.addAll(def.getExcludeUrls());
+            });
+            for (final ExcludeUrl excludeUrl : excludeUrlList) {
+                if (excludeUrl.getUrl().isPresent()) {
+                    final List<IPPollAddress> specifics = new LinkedList<IPPollAddress>();
+                    List<String> ipAddressList = addToSpecificsFromURL(specifics,
+                            excludeUrl.getUrl().get(),
+                            excludeUrl.getForeignSource().orElse(null),
+                            excludeUrl.getLocation().orElse(null),
+                            0,
```

Comment:
we need to get timeout, retries from definition and use that here.  May be add `timeout` and `retries` as transient variables ( not needed for config ) in ExcludeUrl

Reply:
Since this is only for exclude, you don't need to consider `timeout`, `retries`.